vision_agent/agent/vision_agent_v2.py

```diff
@@ -31,6 +33,15 @@
 _CONSOLE = Console()
 
 
+def build_working_memory(working_memory: Mapping[str, List[str]]) -> Sim:
+    data: Mapping[str, List[str]] = {"desc": [], "doc": []}
+    for key, value in working_memory.items():
+        data["desc"].append(key)
+        data["doc"].append("\n".join(value))
+    df = pd.DataFrame(data)  # type: ignore
+    return Sim(df, sim_key="desc")
```

Comment:
Can you describe what happens in this function ? Sim returns a df with embedding calculated on the given column. From what i see we build df from a working memory which contains desc and doc. Are the desc and doc are description and doc string of tools or are they something else ? because i was under the assumption working memory is the all the artifacts from a previous run.

Reply:
Oh yeah, for memory desc and doc are probably not the best terminology. In this case, desc is the subtask string and doc is the debug attempts at trying to build code for that subtask. For example, you might have:

desc: "load the file 'dog.png'"
doc: """
image = open('cat.png')

reflection: You opened the wrong image name, it should be 'dog.png'

image = open('dog.png'
"""

So this context could be saved as long term memory. Then the next time the agent encounters the question "load the file 'dog.png'" it could retrieve this context to help it.